src/docs/arch-snapshot.dox

```diff
@@ -167,4 +170,15 @@ transaction and snapshot state. This being independent from the application's ru
 When checkpoints are running, application transactions using snapshots will also store the checkpoints transaction ID as a concurrent transaction.
 This allowing a snapshot to ignore any uncommitted changes the checkpoint has written to the metadata.
 
+@section snapshot_read_uncommitted Snapshot Read-uncommitted \& Modifies
+
+Snapshot read-uncommitted readers can not reconstruct a full value when reading an update chain that
+contains a modify. The reason is because another transaction with isolation snapshot can abort the
+modifies or updates in parallel to reader, which can unintentionally skip a required modify or
+we are not able to find a base update to apply the modifies on. Effectively read-uncommitted reader
+must read and race with the isolation snapshot write. It is difficult to distinguish if
+an aborted operation happened prior to the call of the function or if it has been done in parallel
+to the reader. Therefore whenever snapshot read-uncommitted readers encounter an update chain with a
+modify, the user will expect a WT_RESTART. Once the user reads the record again, the update
+chain will contain the aborted operations already.
```

Comment:
Wasn't the problem specifically *aborted* modifies? Or is there more to it that I've missed (haven't had time to follow the ticket comments closely - sorry!)

If we're OK with the approach of "any modifies bad", I think this paragraph needs a going over with a grammar comb - happy to do that IRL.

Reply:
Fixed as per discussion